### PR TITLE
add logconfig proto conversions

### PR DESCRIPTION
**Pull request by puranamp**
This change adds proto conversion utilities for LogPatternConfig. We need this for when we want to include the log configuration field within the config for cloud robots, which is currently not being handled in app.

### CODE REVIEW

File: config/proto_conversions.go [hunk 1]
```diff
@@ -929,3 +929,19 @@ func logAnyFragmentOverwriteErrors(logger logging.Logger, overwriteFragmentStatu
 		logger.Errorw("error in overwriting fragment", "error", status.GetError())
 	}
 }
+
+// LogConfigToProto converts a LoggerPatternConfig type to its proto equivalent.
+func LogConfigToProto(logConfig LoggerPatternConfig) (*pb.LogPatternConfig, error) {
```
Comment (benjirewis) [inline config/proto_conversions.go:934]: Can we add a round-trip test to `proto_conversions_test.go`?

**Review by benjirewis — COMMENTED** [review]
Nice! Just a request for a test.

**Review by maximpertsov — CHANGES_REQUESTED** [review]
We should also incorporate this in the top-level `FromProto` method

**Review by maximpertsov — APPROVED** [review]

**Review by benjirewis — APPROVED** [review]

File: config/config.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ type configData struct {
 	Debug               bool                  `json:"debug,omitempty"`
 	DisablePartialStart bool                  `json:"disable_partial_start"`
 	EnableWebProfile    bool                  `json:"enable_web_profile"`
-	LogConfig           []LoggerPatternConfig `json:"log"`
+	LogConfig           []LoggerPatternConfig `json:"log,omitempty"`
```
Comment (benjirewis) [inline config/config.go:84]: Do we not need a `LogConfig` field on the `Config` struct above, too?

**Review by benjirewis — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 2 above)
Comment (puranamp) [reply config/config.go:84]: There is a field [here](https://github.com/viamrobotics/rdk/blob/3b8b07f972a2810b246e9c4e4c94c9247adeba86/config/config.go#L40). Should I move it to the bottom of the struct? I see why it is hard to find

Comment (maximpertsov) [reply config/config.go:84]: It's already added [on line 40](https://github.com/viamrobotics/rdk/pull/4193/files/c5eb8f68da3d55a5819eeae54e222ba9c631951e#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R40), but maybe it's worth adding a comment for it.

Comment (benjirewis) [reply config/config.go:84]: Ah thanks for pointing it out; nope all good, you can leave where is. Just missed it 😮‍💨 .

**Review by benjirewis — APPROVED** [review]

**Pull request merged**